libs/SalesforceSDK/src/com/salesforce/androidsdk/security/SalesforceKeyGenerator.java

```diff
@@ -135,6 +154,47 @@ public static String getSHA256Hash(String privateKey) {
         return hashedString;
     }
 
+    /**
+     * Generate a keypair and get the encoded public key string.
+     *
+     * @param name Alias of the entry in which the generated key will appear in Android KeyStore.
+     * @return RSA public key string.
+     */
+    public static synchronized String getRSAPublicString(String name, int length) {
+        String publicKeyBase64 = null;
+        createRSAKeys(name, length);
+        try {
+            publicKeyBase64 = Base64.encodeToString(loadKeyStore().getCertificate(name).getPublicKey().getEncoded(),
```

Comment:
Same comment as below. I think this line should be called before `createRSAKeys` in the event that it has already been created.